src/container.h

```diff
@@ -92,4 +92,53 @@ class RingDeque {
   T buffer_[CAPACITY];
 };
 
+//Stupid reimplementation of the deque
```

Comment:
Your RingQueue class _looks_ like it does the same thing as the RingDeque class. You just added a queue_size_ member instead of calculating the value based on the back_pos_/front_pos_ (read_pos_/write_pos_ in RingDeque) and added accessors for the back_pos_/fron_pos_t members. It would be cleaner to use the existing class and add the accessors to that class (if they are actually needed).